apps/charterafrica/src/lib/ecosystem/airtable.js

```diff
@@ -1,6 +1,284 @@
+import api from "@/charterafrica/lib/payload";
+import fetchJson, { FetchError } from "@/charterafrica/utils/fetchJson";
+
+const airtable = new Airtable({
+  apiKey: process.env.AIRTABLE_API_TOKEN,
+});
+
+function getter(data, key) {
+  return key ? data?.[key] : null;
+}
+
+async function mapSupportersToFields(supporters, config, tableData) {
+  const {
+    schema: { partnerTableColumns },
+  } = config;
+  const { name, url, logo } = partnerTableColumns;
+  const { partners } = tableData;
+  const mapped = supporters.map((id) => {
+    try {
+      const { fields } = partners.find((item) => id === item.id);
+      return {
+        name: getter(fields, name),
+        website: getter(fields, url),
+        logo: getter(fields, logo),
+      };
+    } catch (error) {
+      Sentry.captureMessage(error.message);
+      return null;
+    }
+  });
+  return mapped.filter(Boolean);
+}
+
+async function getToolsPerAirtableId(ids) {
+  if (!ids || !ids?.length) {
+    return [];
+  }
+  const { docs } = await api.getCollection(TOOL_COLLECTION, {
+    where: {
+      airtableId: {
+        in: ids?.join(","),
+      },
+    },
+  });
+  return docs.map(({ id }) => id);
+}
+
+async function mapSocialMediaToFields(socialMedia, config, tableData) {
+  const {
+    schema: { socialMediaTableColumns },
+  } = config;
+  const { name, url } = socialMediaTableColumns;
+  const { socialMedia: socialMediaData } = tableData;
+  const mapped = socialMedia.map((id) => {
+    try {
+      const { fields } = socialMediaData.find((item) => item.id === id);
+      return {
+        name: getter(fields, name),
+        website: getter(fields, url),
+      };
+    } catch (error) {
+      Sentry.captureMessage(error.message);
+      return null;
+    }
+  });
+  return mapped.filter(Boolean);
+}
+
+export const getListFromAirtable = async ({ baseId, tableIdOrName }) => {
+  const base = airtable.base(baseId);
+  return base(tableIdOrName).select().all();
+};
+
+export const processOrganisationFromAirTable = async (
+  data,
+  config,
+  tableData
+) => {
+  try {
+    const {
+      schema: { organisationTableColumns },
+    } = config;
+    const description = {
+      en: getter(data, organisationTableColumns.description.en),
+      pt: getter(data, organisationTableColumns.description.pt),
+      fr: getter(data, organisationTableColumns.description.fr),
+    };
+    const partners = await mapSupportersToFields(
+      getter(data, organisationTableColumns.partners) || [],
+      config,
+      tableData
+    );
+    const supporters = await mapSupportersToFields(
+      getter(data, organisationTableColumns.supporters) || [],
+      config,
+      tableData
+    );
+    const socialMedia = await mapSocialMediaToFields(
+      getter(data, organisationTableColumns.socialMedia) || [],
+      config,
+      tableData
+    );
+    const tools = await getToolsPerAirtableId(
+      getter(data, organisationTableColumns.tools)
+    );
+    const unLocalizedData = {
```

Comment:
```suggestion
    const commonData = {
```
When you say `unLocalisedData`, do you mean it needs localizing?